training/src/main/scala/com/airbnb/aerosolve/training/pipeline/GenericPipeline.scala

```diff
@@ -509,14 +529,14 @@ object GenericPipeline {
       .map(x => hiveTrainingToExample(x, schema, isMulticlass))
   }
 
-  def isTraining(examples : Example) : Boolean = {
+  def isTraining(frac: Double = DEFAULT_TRAINING_FRACTION)(examples : Example) : Boolean = {
     // Take the hash code mod 255 and keep the first 16 as holdout.
-    (examples.toString.hashCode & 0xFF) > 16
+    (examples.toString.hashCode & 0xFF) > 255 - Math.floor(255 * frac)
   }
```

Comment:
What `toString` on the Example class does, if I am not wrong, is producing something like `Example@[hex hashcode]`. So `examples.toString.hashCode` is like performing hashing twice on the object, which is probably tiny performance loss. In terms of correctness, this might not be too big of an issue but I would expect Thrift object hashcode to be more uniformly distributed ever so slightly.